packages/story-editor/src/utils/checkResourceAccess.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Check if the resource can be accessed directly.
+ *
+ * Makes a HEAD request, which in turn triggers a CORS preflight request
+ * in the browser.
+ *
+ * If the request passes, we don't need to do anything.
+ * If it doesn't, it means we need to run the resource through our CORS proxy at all times.
+ *
+ * @type {function(): boolean}
+ */
+async function checkResourceAccess(link) {
```

Comment:
My recommendation was to move this function within `useCORSProxy` hook as it's related to that. Whats' the rationale for making this a standalone util function?

Reply:
You didn't make a recommendation. There is no comment on this PR. It makes sense a util, makes it re-usable.

Reply:
To be clear, we need this as a function, as need to value from it to cache it, in the resource, link and page attachment.

Reply:
> You didn't make a recommendation. There is no comment on this PR. It makes sense a util, makes it re-usable.

https://github.com/google/web-stories-wp/pull/9713#discussion_r750198995

Reply:
> To be clear, we need this as a function, as need to value from it to cache it, in the resource, link and page attachment.

I don't understand how moving this function to a different file causes problems with any of this.

Reply:
I don't see the point of it is living in that function. You just like it there?

Reply:
Checking resource access for proxying and getting the proxy URL are coupled, you wouldn't use one without the other. It makes sense for this functionality to live together.

Reply:
Moved.